app/src/main/java/nl/eduvpn/app/service/SecurityService.kt

```diff
@@ -0,0 +1,177 @@
+/*
+ *  This file is part of eduVPN.
+ *
+ *     eduVPN is free software: you can redistribute it and/or modify
+ *     it under the terms of the GNU General Public License as published by
+ *     the Free Software Foundation, either version 3 of the License, or
+ *     (at your option) any later version.
+ *
+ *     eduVPN is distributed in the hope that it will be useful,
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *     GNU General Public License for more details.
+ *
+ *     You should have received a copy of the GNU General Public License
+ *     along with eduVPN.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package nl.eduvpn.app.service
+
+import android.content.Context
+import android.util.Base64
+import androidx.annotation.CheckResult
+import androidx.annotation.VisibleForTesting
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import nl.eduvpn.app.BuildConfig
+import nl.eduvpn.app.utils.Log
+import org.libsodium.jni.NaCl
+import org.libsodium.jni.Sodium
+import java.nio.charset.StandardCharsets
+import java.security.SecureRandom
+
+/**
+ * Service which is responsible for all things related to security.
+ *
+ *
+ * Created by Daniel Zolnai on 2017-08-01.
+ */
+class SecurityService(private val context: Context) {
+    companion object {
+        private val TAG = SecurityService::class.java.name
+
+        // Public key ID is used as map keys, public key bytes are used as map values.
+        private lateinit var MINISIGN_PUBLIC_KEYS: Map<String, ByteArray> //todo: val
```

Comment:
This can only be a val if you make it a MutableMap.
But I'm also fine with a lateinit var.